src/main.cpp

```diff
@@ -432,20 +432,22 @@ class bearer_token_eviction_manager
 
             log::trace("Evicting expired bearer tokens ...");
             irods::http::process_stash::erase_if([](const auto& _k, const auto& _v) {
-                try {
-                    const auto* p = boost::any_cast<const irods::http::authenticated_client_info>(&_v);
-                    const auto erase_token = (p && std::chrono::steady_clock::now() >= p->expires_at);
+			  // Check for client bearer token
+			  const auto* client_info{boost::any_cast<const irods::http::authenticated_client_info>(&_v)};
+			  const auto erase_token{(client_info && std::chrono::steady_clock::now() >= client_info->expires_at)};
 
-                    if (erase_token) {
-                        log::debug("Evicted bearer token [{}].", _k);
-                    }
+			  // Check for OAuth 2.0 state param
+			  const auto* expire_time{boost::any_cast<const std::chrono::steady_clock::time_point>(&_v)};
+			  const auto erase_state{(expire_time && std::chrono::steady_clock::now() >= *expire_time)};
 
-                    return erase_token;
-                }
-                catch (...) {
-                }
+			  // Determine if value need to be deleted
+			  const auto erase_value{erase_token || erase_state};
 
-                return false;
+			  if (erase_value) {
+				log::debug("Evicted token [{}].", _k);
```

Comment:
This message could be adjusted based on the booleans on L444.

Being more specific here would help the admin if they ever needed to confirm eviction of something.

Reply:
Done.